### PR TITLE
feat(constraints-api): add startup configuration

### PR DESCRIPTION
**Pull request by thedevbirb**
Adds the `ConstraintsApiConfig` struct that allows disabling some signature checks if needed.

### CODE REVIEW

File: crates/common/src/config.rs [hunk 1]
```diff
@@ -125,6 +127,20 @@ pub enum NetworkConfig {
     },
 }
 
+/// Configuration for the [Constraints API](https://docs.boltprotocol.xyz/technical-docs/api/builder)
+///
+/// These checks are enabled by default. Disabling them is recommended only for testing purposes.
+#[derive(Serialize, Deserialize, Clone)]
+pub struct ConstraintsApiConfig {
+    pub check_constraints_signature: bool,
```
Comment (merklefruit) [inline crates/common/src/config.rs:138]: can you add a doc comment explaining that this removes the sig verification check on constraints? 
It's almost obvious but better to be clear.

**Review by merklefruit — APPROVED** [review]
LGTM

File: crates/common/src/config.rs (hunk 1 above)
Comment (thedevbirb) [reply crates/common/src/config.rs:138]: Done, lmk if it's okay!

Comment (merklefruit) [reply crates/common/src/config.rs:138]: Yesss

**Pull request merged**